src/react-vite/12-testing/testing.md

```diff
@@ -38,71 +38,77 @@ React Testing Library’s north star is:
-
-### Rendering and verifying a component in a test
+rather than the implementation details of the components.
 
-Let’s take a look at some code that we added in [Handling User Inputs and
-Forms](./controlled-vs-uncontrolled.html).
+### <span id="rendering-and-verifying">Rendering and verifying a component in a test</span>
 
-@sourceref ../../../exercises/react-vite/11-controlled-vs-uncontrolled/03-problem/src/components/FormTextField/FormTextField.test.tsx
-@highlight 10-15, only
+OK, you're convinced that testing-library is a good idea, how do you use it to create tests for your
+React components?
 
-Each test consists of arguments to the Vite provided test function, `it`. The first argument is a
-short description of what the test is examining. The convention is that the description string takes
-"it" as a prefix and proceeds from there, e.g. "(it) renders with correct label and type."
+Let's take a look at an example; say we want to test a component we created named `FormTextField`
+that renders a form field. We expect that the component will generate HTML that looks like this:
 
-The second argument is a callback function that runs the test. In the example above the first line
-of the callback invokes `render` and passes it a single argument, JSX for the `FormTextField`
-component including props. After `render` completes the `screen` (a test DOM) will contain the
-elements created by our React code.
+```html
+<div>
+  <label htmlFor="inputId">Phone:</label>
+  <input id="inputId" type="tel" value="555-555-5555" />
+</div>
+```
 
-@sourceref ../../../exercises/react-vite/11-controlled-vs-uncontrolled/03-problem/src/components/FormTextField/FormTextField.test.tsx
-@highlight 11, only
+We want to test `FormTextField` to make sure it generates the DOM we expect. If you're already
+familiar with testing frontend JavaScript code the following pattern will probably be recognizable:
+each test consists of arguments to the Vite provided test function, `it`. The first argument is a
+short description of what the test is verifying. The convention is that the description string takes
+"it" as a prefix and proceeds from there, e.g. "[it] renders the correct label and value"
 
-Now it’s time to see if the DOM matches what we expected to create. The React Testing Library
-provides the `screen` object that allows us to select elements from the DOM. In the current scenario
-we’ll use `getByLabelText` which returns the `<input>` associated by `id` with a single `<label>`
-that has the text "Test Label". You may have intuited that `getByLabelText` accepts a string, but it
-also accepts a regex, in this case one that matches any part of the label text and ignores case.
+The second argument to `it` is a callback function that is called to run the test. Inside the
+callback invoke the testing-library function `render` and pass it a single argument, JSX for your
```

Comment:
`callback invoke` -> `callback, invoke`

Reply:
Also, maybe `argument, JSX` -> `argument: the JSX`

Reply:
fixed in next PR